src/vulkan-renderer/wrapper/device.cpp

```diff
@@ -311,74 +311,88 @@ Device::~Device() {
-    name_info.object = object;
-    name_info.pObjectName = name.c_str();
+        auto name_info = make_info<VkDebugMarkerObjectNameInfoEXT>();
+        name_info.objectType = type;
+        name_info.object = reinterpret_cast<std::uint64_t>(object);
+        name_info.pObjectName = name.c_str();
 
-    m_vk_debug_marker_set_object_name(m_device, &name_info);
+        if (m_vk_debug_marker_set_object_name(m_device, &name_info) != VK_SUCCESS) {
+            throw std::runtime_error("Failed to assign Vulkan debug marker name to Vulkan resource " + name + "!");
+        }
+    }
 }
 
 void Device::set_object_tag(const std::uint64_t object, const VkDebugReportObjectTypeEXT type, const std::uint64_t name,
-                            const std::size_t tag_size, const void *tag) {
-    assert(m_device);
-    assert(name);
-    assert(tag_size > 0);
-    assert(tag);
-    assert(m_vk_debug_marker_set_object_tag);
-
-    auto tagInfo = make_info<VkDebugMarkerObjectTagInfoEXT>();
-    tagInfo.objectType = type;
-    tagInfo.object = object;
-    tagInfo.tagName = name;
-    tagInfo.tagSize = tag_size;
-    tagInfo.pTag = tag;
-
-    m_vk_debug_marker_set_object_tag(m_device, &tagInfo);
+                            const std::size_t tag_size, const void *tag) const {
+    if (m_enable_vulkan_debug_markers) {
+        assert(m_device);
+        assert(name);
+        assert(tag_size > 0);
+        assert(tag);
+        assert(m_vk_debug_marker_set_object_tag);
+
+        auto tagInfo = make_info<VkDebugMarkerObjectTagInfoEXT>();
+        tagInfo.objectType = type;
+        tagInfo.object = object;
+        tagInfo.tagName = name;
+        tagInfo.tagSize = tag_size;
+        tagInfo.pTag = tag;
+
+        if (m_vk_debug_marker_set_object_tag(m_device, &tagInfo) != VK_SUCCESS) {
+            throw std::runtime_error("Failed to assign Vulkan debug marker data tag to Vulkan resource!");
+        }
+    }
 }
 
 void Device::bind_debug_region(const VkCommandBuffer command_buffer, const std::string &name,
-                               const std::array<float, 4> color) {
-    assert(command_buffer);
-    assert(!name.empty());
-    assert(m_vk_cmd_debug_marker_begin);
-
-    auto debug_marker = make_info<VkDebugMarkerMarkerInfoEXT>();
-    debug_marker.color[0] = color[0];
-    debug_marker.color[1] = color[1];
-    debug_marker.color[2] = color[2];
-    debug_marker.color[3] = color[3];
-    debug_marker.pMarkerName = name.c_str();
-
-    m_vk_cmd_debug_marker_begin(command_buffer, &debug_marker);
+                               const std::array<float, 4> color) const {
+    if (m_enable_vulkan_debug_markers) {
+        assert(command_buffer);
+        assert(!name.empty());
+        assert(m_vk_cmd_debug_marker_begin);
+
+        auto debug_marker = make_info<VkDebugMarkerMarkerInfoEXT>();
+        debug_marker.color[0] = color[0];
+        debug_marker.color[1] = color[1];
+        debug_marker.color[2] = color[2];
+        debug_marker.color[3] = color[3];
```

Comment:
We could use std::copy. Similiar: #215  (not 100% sure if its possible)
Same below.

Reply:
Done.